include/circt/Dialect/Calyx/CalyxOps.h

```diff
@@ -37,14 +37,35 @@ class ControlLike : public mlir::OpTrait::TraitBase<ConcreteType, ControlLike> {
   }
 };
 
-/// The direction of a Calyx port.
-enum PortDirection { INPUT = 0, OUTPUT = 1 };
+/// The port direction attribute follows the implementation style of FIRRTL
+/// module port direction attributes.
+enum Direction { Input = 0, Output };
+namespace direction {
+
+/// The key in a components's attribute dictionary used to find the direction.
+constexpr const char *attrKey = "portDirections";
+
+/// Return an output direction if \p isOutput is true, otherwise return an
+/// input direction.
+Direction get(bool isOutput);
```

Comment:
Style comment: I somewhat prefer a static method like `Direction::get` rather than a function in the namespace like `direction::get`. The former is more in line with how MLIR provides helpers for things like attributes (e.g. `ArrayAttr::get`). Up to you though.

Reply:
I understand - I didn't pay careful attention to this code since it [essentially is a copy of what is currently done in FIRRTL](https://github.com/llvm/circt/blob/main/include/circt/Dialect/FIRRTL/FIRRTLOps.h#L38).

My hunch would be to leave it as is. This would put it in the scope of anyone in the future refactoring this and the FIRRTL code to a single, shared implementation.